### DISCUSSION TITLE
Aerostructural-MACH tutorial wing

### DISCUSSION THREAD

**Discussion by hxl1310** [mdolab/dafoam#617] [Tutorials]
When I completed the Aerostructural Mach tutorial wing in the tutorial, I couldn't find the optimized bdf file and didn't know how to view the stress cloud map and skin thickness of the optimized wing.

**Comment by friedenhe** [comment]
Please check the slides and video from the 2022 workshop. https://dafoam.github.io/mydoc_workshops.html

**Reply by hxl1310** [reply]
Thank you very much！

**Reply by hxl1310** [reply]
After running the Aerostructural Mach tutorial wing as a basic case, the optimized structural weight and skin thickness information cannot be found. After watching the video you mentioned, do we need to add structural weight and skin thickness to the objective function or constraints in order to obtain relevant optimization results.
Does the basic case not include optimization of structural weight and skin thickness?
Does the following information appear in the logOPt file after running the basic case affect the optimization of structural quality?

/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_pre.distributor' <class TacsDVComp>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.coupling.aero.solver' <class DAFoamSolver>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_post.eval_funcs' <class TacsFunctions>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_post.mass_funcs' <class MassFunctions>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/visualization/opt_report/opt_report.py:634: UserWarning: Attempting to set identical low and high ylims makes transformation singular; automatically expanding.
  ax.set_ylim([ymin_plot, ymax_plot])
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_pre.distributor' <class TacsDVComp>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.coupling.aero.solver' <class DAFoamSolver>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_post.eval_funcs' <class TacsFunctions>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_post.mass_funcs' <class MassFunctions>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/visualization/opt_report/opt_report.py:634: UserWarning: Attempting to set identical low and high ylims makes transformation singular; automatically expanding.
  ax.set_ylim([ymin_plot, ymax_plot])
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_pre.distributor' <class TacsDVComp>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.coupling.aero.solver' <class DAFoamSolver>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_post.eval_funcs' <class TacsFunctions>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_post.mass_funcs' <class MassFunctions>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/visualization/opt_report/opt_report.py:634: UserWarning: Attempting to set identical low and high ylims makes transformation singular; automatically expanding.
  ax.set_ylim([ymin_plot, ymax_plot])
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_pre.distributor' <class TacsDVComp>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.coupling.aero.solver' <class DAFoamSolver>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_post.eval_funcs' <class TacsFunctions>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_post.mass_funcs' <class MassFunctions>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/visualization/opt_report/opt_report.py:634: UserWarning: Attempting to set identical low and high ylims makes transformation singular; automatically expanding.
  ax.set_ylim([ymin_plot, ymax_plot])
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_pre.distributor' <class TacsDVComp>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.coupling.aero.solver' <class DAFoamSolver>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_post.eval_funcs' <class TacsFunctions>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_post.mass_funcs' <class MassFunctions>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/visualization/opt_report/opt_report.py:634: UserWarning: Attempting to set identical low and high ylims makes transformation singular; automatically expanding.
  ax.set_ylim([ymin_plot, ymax_plot])
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_pre.distributor' <class TacsDVComp>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.coupling.aero.solver' <class DAFoamSolver>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_post.eval_funcs' <class TacsFunctions>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/component.py:140: OMDeprecationWarning:'cruise.struct_post.mass_funcs' <class MassFunctions>: It appears this component mixes distributed/non-distributed inputs and outputs, so it may break starting with OpenMDAO 3.25, where the convention used when passing data between distributed and non-distributed inputs and outputs within a matrix free component will change. See https://github.com/OpenMDAO/POEMs/blob/master/POEM_075.md for details.
/home/hxl/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/visualization/opt_report/opt_report.py:634: UserWarning: Attempting to set identical low and high ylims makes transformation singular; automatically expanding.
  ax.set_ylim([ymin_plot, ymax_plot])

**Comment by friedenhe** [comment]
> Does the basic case not include optimization of structural weight and skin thickness?

Right, the tutorial does not include structural weight. But it does have skin thickness (constant t=3 mm). The skin, ribs, and spar thicknesses are not design variables and remain constant during optimization. The message you posted is a warning, not an error.
